### PR TITLE
Patreon

### PR DESCRIPTION
**Pull request by Drapersniper**
Implements #8  and #7

### CODE REVIEW

**Comment by kennnyshiwa** [comment]
@Drapersniper so seems to work well so far, however in testing i got it to create an item but then can't equip it even though has_patreon is true and i still have the role

File: adventure/adventure.py [hunk 1]
```diff
@@ -2277,64 +2329,145 @@ def add_stats(word_stats):
...
+        """Sync users with Patreon"""
+        if not ctx.guild or ctx.guild.id != 489162733791739950:
+            return await smart_embed(ctx, ("This command must be run in the BB-8 Support Server"))
+        patreon_role = ctx.guild.get_role(575723093843116070)
+        if patreon_role is None:
+            return await smart_embed(ctx, ("Patreon Role doesn't exist"))
+        time_now = time.time()
+        for member in ctx.guild.members:
+            await asyncio.sleep(0)
+            if patreon_role in member.roles:
```
Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
            if patreon_role in member.roles:
                data = await self.config.user(member).patron.all()
                data["has_patron"] = True
                data["first_patron"] = int(time.time())
                await self.config.user(member).patron.set(data)
```

File: adventure/adventure.py [hunk 2]
```diff
@@ -2277,64 +2329,145 @@ def add_stats(word_stats):
...
+                with self.config.user(member).patron.all() as patron_data:
```
Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
```

File: adventure/adventure.py [hunk 3]
```diff
@@ -2277,64 +2329,145 @@ def add_stats(word_stats):
...
+                    if not patron_data["has_patron"]:
```
Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
```

File: adventure/adventure.py [hunk 4]
```diff
@@ -2277,64 +2329,145 @@ def add_stats(word_stats):
...
+                        patron_data["has_patron"] = True
```
Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
```

File: adventure/adventure.py [hunk 5]
```diff
@@ -2277,64 +2329,145 @@ def add_stats(word_stats):
...
+                        patron_data["first_patron"] = time_now
```
Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
```

Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
```

File: adventure/adventure.py [hunk 6]
```diff
@@ -2277,64 +2329,145 @@ def add_stats(word_stats):
...
+        patreon_role = ctx.guild.get_role(575723093843116070)
+        if patreon_role is None:
+            return await smart_embed(ctx, ("Patreon Role doesn't exist"))
+        time_now = time.time()
+        for member in ctx.guild.members:
+            await asyncio.sleep(0)
+            if patreon_role in member.roles:
+                data = await self.config.user(member).patron.all()
+                data["has_patron"] = True
+                data["first_patron"] = int(time.time())
```
Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
                data["first_patron"] = int(time_now)
```

Comment (Drapersniper) [inline adventure/adventure.py]:
```suggestion
                data["first_patron"] = int(time_now)
```

**Pull request merged**